api/tests/unit/domain/usecases/assign-certification-officer-to-session_test.js

```diff
@@ -3,62 +3,97 @@ const assignCertificationOfficerToJurySession = require('../../../../lib/domain/
-  });
-
   context('when session id is not a number', () => {
-
     it('should throw a NotFound error', async () => {
       // given
-      sessionId = 'notANumber';
+      const jurySessionRepository = {
+        assignCertificationOfficer: sinon.stub(),
+      };
+      const sessionId = 'notANumber';
+      const certificationOfficerId = 1;
 
       // when
-      const error = await catchErr(assignCertificationOfficerToJurySession)({ sessionId, certificationOfficerId, jurySessionRepository });
+      const error = await catchErr(assignCertificationOfficerToJurySession)({
+        sessionId,
+        certificationOfficerId,
+        jurySessionRepository,
+      });
 
       // then
       expect(error).to.be.an.instanceOf(ObjectValidationError);
     });
   });
 
   context('when session id is a number', () => {
-
-    beforeEach(() => {
-      sessionId = 1;
-    });
-
     context('when certificationOfficerId is not a number', () => {
-
       it('should throw a NotFound error', async () => {
         // given
-        certificationOfficerId = 'notANumber';
+        const jurySessionRepository = {
+          assignCertificationOfficer: sinon.stub(),
+        };
+        const sessionId = 1;
+        const certificationOfficerId = 'notANumber';
 
         // when
-        const error = await catchErr(assignCertificationOfficerToJurySession)({ sessionId, certificationOfficerId, jurySessionRepository });
+        const error = await catchErr(assignCertificationOfficerToJurySession)({
+          sessionId,
+          certificationOfficerId,
+          jurySessionRepository,
+        });
 
         // then
         expect(error).to.be.an.instanceOf(ObjectValidationError);
       });
     });
 
     context('when certificationOfficerId is a number', () => {
-      const returnedSession = Symbol('returnedSession');
-
-      beforeEach(() => {
-        certificationOfficerId = 2;
-        jurySessionRepository.assignCertificationOfficer.withArgs({ id: sessionId, assignedCertificationOfficerId: certificationOfficerId }).resolves(returnedSession);
-      });
-
       it('should return the session after assigningUser to it', async () => {
+        // given
+        const returnedSession = Symbol('returnedSession');
+        const sessionId = 1;
+        const certificationOfficerId = 2;
+
+        const jurySessionRepository = {
+          assignCertificationOfficer: sinon.stub(),
+        };
+        jurySessionRepository.assignCertificationOfficer
+          .withArgs({
+            id: sessionId,
+            assignedCertificationOfficerId: certificationOfficerId,
+          })
+          .resolves(returnedSession);
+
+        const finalizedSessionRepository = {
+          assignCertificationOfficer: sinon.stub(),
+        };
+
+        const userRepository = { get: sinon.stub() };
+        userRepository.get
+          .withArgs(2)
+          .resolves({ firstName: 'Severus', lastName: 'Snape' });
```

Comment:
**Comments:** Avec les withArgs, on ne devrait pas avoir a verifier les to.have.been.called 
Si on renvoi une valeur UNIQUEMENT selon des parametre specifique, le seul `expect(actualSession).to.equal(returnedSession);` devrait etre suffisant